code/weapon/weapons.cpp

```diff
@@ -4188,32 +4200,25 @@ void detonate_nearby_missiles(object* killer_objp, object* missile_objp)
  */
 void find_homing_object(object *weapon_objp, int num)
 {
-	object      *objp, *old_homing_objp;
-	weapon_info *wip;
-	weapon      *wp;
-    ship        *sp;
-    ship_info   *sip;
-	float       best_dist;
-    int         homing_object_team;
-    float       dist;
-    float       dot;
-    vec3d       vec_to_object;
     ship_subsys *target_engines = NULL;
 
-	wp = &Weapons[num];
+	weapon* wp = &Weapons[num];
```

Comment:
You could use `auto` here and on the other declarations.  This is by no means required, I just wanted to make sure you were aware of it.

Reply:
I am, yeah. In general I tend to prefer explicit types unless its some obnoxiously long C++ bullcrap.